pkg/webhook/pod/mutating/sidecarset.go

```diff
@@ -393,13 +395,22 @@ func buildSidecars(isUpdated bool, pod *corev1.Pod, oldPod *corev1.Pod, matchedS
 				// merged Env from sidecar.Env and transfer envs
 				initContainer.Env = util.MergeEnvVar(initContainer.Env, transferEnvs)
 				isInjecting = true
-				sidecarInitContainers = append(sidecarInitContainers, initContainer)
+
+				// when sidecar container UpgradeStrategy is HotUpgrade
+				if sidecarcontrol.IsSidecarContainer(initContainer.Container) && sidecarcontrol.IsHotUpgradeContainer(initContainer) {
+					hotContainers, annotations := injectHotUpgradeContainers(hotUpgradeWorkInfo, initContainer)
+					sidecarInitContainers = append(sidecarInitContainers, hotContainers...)
+					for k, v := range annotations {
+						injectedAnnotations[k] = v
+					}
+				} else {
+					sidecarInitContainers = append(sidecarInitContainers, initContainer)
+				}
 			}
 			//process imagePullSecrets
 			sidecarSecrets = append(sidecarSecrets, sidecarSet.Spec.ImagePullSecrets...)
 		}
 
-		sidecarList := sets.NewString()
 		//process containers
```

Comment:
maybe an TODO, implement PodInjectPolicy for initContainers, currently initcontainers only use AfterAppContainer